dotnet/src/Skills/Skills.Web/WebFileDownloadSkill.cs

```diff
@@ -37,6 +37,19 @@ public WebFileDownloadSkill(ILogger<WebFileDownloadSkill>? logger = null)
         this._httpClient = new HttpClient(this._httpClientHandler);
     }
 
+    public async Task DownloadToFileAsync(string source, IDictionary<string, string> headers, SKContext context)
+    {
+        this._httpClient.DefaultRequestHeaders.Clear();
```

Comment:
Just wondering...  Are there any headers in there by default before calling this method?  If so, we should save that collection of headers and re-apply it at the end of this method.

Reply:
TS•Pro

Reply:
Sorry, I was two weeks on vacation. That is a good suggestion, I don't think are there any headers since the httpclient is initialized in the constructor, but maybe to prevent future contamination.

Reply:
@glahaye on a second thought, because _httpClient is private readonly, the class has total control over the object, including the headers. In the same note, when I was doing this PR I thought a cleaner implementation would be that this class receives the httpClient in the constructor (to avoid dependency injection) or with some kind of DI pattern (including maybe accessors), that way, we don't even need to modify this class, and the caller (ex. GitHubSkill) would need to just add the headers for authorization or others. It would have required to modify substantially the current implementation, so I avoided to do that thinking would be better to have a short and sweet PR instead of a refactor one.

I will close this thread since it would be better to refactor this class, but that was out of the scope of the purpose of this PR, if you feel strongly about it, I can close this.